### PR TITLE
Integration test for Service Provider Pagination Issue

### PR DESCRIPTION
**Pull request by ZiyamSanthosh**
This test focuses on the issue https://github.com/wso2/product-is/issues/14428

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/application/mgt/ServiceProviderUserRoleValidationTestCase.java [hunk 1]
```diff
@@ -136,14 +164,23 @@ private void updateAdminUser() throws Exception {
...
-        Assert.assertEquals(count, APPLICATION_COUNT_WITHOUT_FILTER,
-                String.format("The expected application count without a filter does not match the actual application " +
-                                "count: Expected Application Count: %d Actual Application Count: %d.",
-                        APPLICATION_COUNT_WITHOUT_FILTER, count)
-        );
+        if (enableRoleValidation) {
+            Assert.assertEquals(count, APPLICATION_COUNT_WITHOUT_FILTER,
+                    String.format("The expected application count without a filter does not match the actual " +
+                                    "application count: Expected Application Count: %d Actual Application Count: %d.",
+                            APPLICATION_COUNT_WITHOUT_FILTER, count)
```
Comment (ZiyamSanthosh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/application/mgt/ServiceProviderUserRoleValidationTestCase.java:171]: Check indentations

Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/application/mgt/ServiceProviderUserRoleValidationTestCase.java]: Fixed in 57dd13533e381e285ed6d5ad4b5bc7aaf10452d9

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 2]
```diff
@@ -117,7 +117,6 @@
             <class name="org.wso2.identity.integration.test.oauth2.dcrm.api.OAuthDCRMTestCase"/>
             <class name="org.wso2.identity.integration.test.functions.library.mgt.FunctionLibraryManagementTestCase"/>
             <class name="org.wso2.identity.integration.test.oauth2.OAuth2TokenRevocationWithRevokedAccessToken" />
```
Comment (inthirakumaaran) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml:119]: why we are removing these test cases?

Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Fixed in 010ae9d4a9657caee9ad2ef796c693143c93e6be

**Review by inthirakumaaran — APPROVED** [review]

**Pull request merged**